execution/block-partitioner/src/v2/state.rs

```diff
@@ -0,0 +1,349 @@
+// Copyright © Aptos Foundation
+
+use crate::{
```

Comment:
Few high level thoughts 

1. Can you take a closer look into places where we are cloning the Transaction or the `StateKey` and see if we can avoid them by dealing with references instead? I have commented in few places but there might be more.

2. Wherever possible, use `par_iter()` with `min_length` to avoid rayon spawning very small task for processing. This can be adverse effect on the performance.

Reply:
1. fixed. Clone is only needed as when initially numbering them.
2. Will find another time to bench it and decide the right threshold. Maybe another PR?